packages/gatsby/src/utils/node-manifest.ts

```diff
@@ -60,21 +60,51 @@ const NODE_MANIFEST_FILE_LIMIT = getNodeManifestFileLimit()
  */
 async function findPageOwnedByNode({
   nodeId,
+  fullNode,
   slug,
 }: {
   nodeId: string
+  fullNode: IGatsbyNode
   slug?: string
 }): Promise<{
   page: INodeManifestPage
   foundPageBy: FoundPageBy
 }> {
   const state = store.getState()
-  const { pages, nodes } = state
-  const { byNode } = state.queries
-
-  // the default page path is the first page found in
-  // node id to page query tracking
-  let pagePath = byNode?.get(nodeId)?.values()?.next()?.value
+  const { pages, nodes, staticQueryComponents } = state
+  const { byNode, byConnection, trackedComponents } = state.queries
+
+  const nodeType = fullNode?.internal?.type
+
+  const firstPageWithNodeAsDataDependency =
```

Comment:
is this a full page object that gets returned? and firstPagePathWithNodeInGraphQLListField is a string?

Reply:
or should it be firstPagePathWithNodeAsDataDependency?

Reply:
yes indeed 😄